File: anagram/anagram.py
def is_anagram(word1, word2):
  if type(word1) is not str or type(word2) is not str:
    raise TypeError('Argumenta must be a string')
  word1 = word1.lower()
  word2 = word2.lower()
  return sorted(word1) == sorted(word2)

print(is_anagram('cat', 'act'))
print(is_anagram('cat', 'dog'))
print(is_anagram(123, []))

File: anagram/anagram.js
function isAnagram(word1, word2) {
  if (typeof word1 !== 'string' || typeof word2 !== 'string') {
    throw TypeError('Arguments must be a string');
  }
  let w1 = word1.toLowerCase().split('').toSorted().join('');
  let w2 = word2.toLowerCase().split('').toSorted().join('');
  return w1 == w2;
}

console.log(isAnagram('cat', 'act')); // true
console.log(isAnagram('cat', 'dog')); // false
console.log(isAnagram(123, [])) // TypeError: Arguments must be a string